File: android-mma-recomp/app/src/main/java/com/example/mmarecomp/ui/components/TargetVsActualBar.kt
package com.example.mmarecomp.ui.components

import androidx.compose.foundation.background
import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Box
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.Spacer
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.height
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material3.MaterialTheme
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.unit.dp

@Composable
fun TargetVsActualBar(
    label: String,
    actual: Double,
    target: Double,
    unit: String,
    modifier: Modifier = Modifier,
) {
    val ratio = if (target > 0) (actual / target).coerceAtMost(1.2) else 0.0
    // Jamais rouge / culpabilisant : au pire un orange doux si largement dépassé.
    val barColor = if (ratio > 1.15) MaterialTheme.colorScheme.secondary else MaterialTheme.colorScheme.primary

    Column(modifier = modifier.fillMaxWidth()) {
        Row(modifier = Modifier.fillMaxWidth(), horizontalArrangement = Arrangement.SpaceBetween) {
            Text(label, style = MaterialTheme.typography.bodyMedium)
            Text(
                "${actual.toInt()}/${target.toInt()} $unit",
                style = MaterialTheme.typography.bodySmall,
                color = MaterialTheme.colorScheme.onSurfaceVariant,
            )
        }
        Spacer(Modifier.padding(top = 4.dp))
        Box(
            modifier = Modifier
                .fillMaxWidth()
                .height(7.dp)
                .clip(RoundedCornerShape(999.dp))
                .background(MaterialTheme.colorScheme.surfaceVariant),
        ) {
            Box(
                modifier = Modifier
                    .fillMaxWidth(ratio.toFloat().coerceIn(0f, 1f))
                    .height(7.dp)
                    .clip(RoundedCornerShape(999.dp))
                    .background(barColor),
            )
        }
    }
}

File: ios-mma-recomp/MMARecomp/Views/Components/TargetVsActualBar.swift
import SwiftUI

struct TargetVsActualBar: View {
    let label: String
    let actual: Double
    let target: Double
    let unit: String

    private var ratio: Double {
        guard target > 0 else { return 0 }
        return min(actual / target, 1.2)
    }

    private var barColor: Color {
        // Jamais rouge / culpabilisant : au pire un orange doux si largement dépassé.
        ratio > 1.15 ? .orange : .accentColor
    }

    var body: some View {
        VStack(alignment: .leading, spacing: 4) {
            HStack {
                Text(label).font(.subheadline)
                Spacer()
                Text("\(Int(actual))/\(Int(target)) \(unit)")
                    .font(.caption)
                    .foregroundStyle(.secondary)
            }
            GeometryReader { geo in
                ZStack(alignment: .leading) {
                    RoundedRectangle(cornerRadius: 6)
                        .fill(Color.secondary.opacity(0.15))
                    RoundedRectangle(cornerRadius: 6)
                        .fill(barColor)
                        .frame(width: geo.size.width * min(ratio, 1.0))
                }
            }
            .frame(height: 8)
        }
    }
}
